tree-sitter/src/TreeSitter/Unmarshal.hs

```diff
@@ -218,9 +218,9 @@ getFields = go Map.empty -- >>= \fields -> liftIO (print (Map.keys fields)) >> p
               fieldName <- peekFieldName
               keepGoing <- step
               let fs' = case fieldName of
-                    Just fieldName' -> Map.insertWith (++) fieldName' [node'] fs
+                    Just fieldName' -> Map.insertWith (flip (++)) fieldName' [node'] fs
                     _ -> if nodeIsNamed node' /= 0
-                      then Map.insertWith (++) (FieldName "extraChildren") [node'] fs
+                      then Map.insertWith (flip (++)) (FieldName "extraChildren") [node'] fs
```

Comment:
`insertWith` applies its argument in the exact opposite order to e.g. `unionWith` or my intuition, which meant we were always adding new nodes to the front of the list.